src/vizy/textingdialog.py

```diff
@@ -0,0 +1,116 @@
+        dialog = Kdialog(title=[Kritter.icon("commenting"), "Texting Bot Configuration"], layout=layout)
+
+        self.layout = KsideMenuItem("Texting", dialog, "commenting") 
+        
+        # Get and run state of dialog
+        self.update_state() # set token for first time, ensuring proper display
+
+        @dialog.callback_view()
+        def func(open):
+            if open:
+                return self.update_state() # Entering -- update GUI state.
+
+        @self.submit_token.callback(self.token_text.state_value())
+        def func(token):
+            if not callback_context.client.authentication&pmask:
+                return
+            self.kapp.push_mods(self.submit_token.out_spinner_disp(True))
+            mods = self.submit_token.out_spinner_disp(False) 
+            try:
+                self.texting_client.set_token(token)
+            except Exception as e:
+                return mods + self.status.out_value(f"There has been an error: {e}")
+            return mods + self.update_state()
+                
+        @self.remove_token.callback()
+        def func():
+            if not callback_context.client.authentication&pmask:
+                return
+            try:
+                self.texting_client.remove_token()
+            except Exception as e:
+                return self.status.out_value(f"There has been an error: {e}")
+            return self.update_state()
+
+        @self.subscriber_select.callback()
+        def func(selection):
+            self.subscriber_selection = selection
+            disabled = not bool(selection)
+            return self.delete_button.out_disabled(disabled)
+
+        @self.delete_button.callback()
+        def func():
+            return self.delete_text.out_value(f'Are you sure you want to delete "{self.subscriber_selection}" subscriber from list?') + self.delete_subscriber_yesno.out_open(True)
+
+        @self.delete_subscriber_yesno.callback_response()
+        def func(val):
+            # remove subscriber from recipient list where user's name is key
+            if val:
+                # find id associated with username
+                userid = [id for (id, name) in self.subscribers.items() if name == self.subscriber_selection]
+                userid = userid[0]              # unwrap id
+                del self.subscribers[userid]    # delete key from subscribers
+                self.text_visor.config['subscribers'] = self.subscribers    # update subscriber list
+                self.text_visor.config.save()                               # save list to file
+                self.kapp.push_mods(self.subscriber_select.out_value(''))   # clear output
+                return self.update_state()
+
+    def update_state(self):
+        # update subscriber list
+        self.text_visor.config.load()
+        self.subscribers = self.text_visor.config['subscribers']
+        self.kapp.push_mods(self.subscriber_select.out_options(self.subscribers.values()))
```

Comment:
You could just add self.subscriber_select.out_options(self.subscribers.values()) to the result (below).  If you call push_mods, it will result in a separate message back to the client.

Reply:
I don't think you need to do config.load() here (L107)